modules/op-overview.adoc

```diff
@@ -0,0 +1,98 @@
+// This module is included in the following assembly:
+//
+// *openshift_pipelines/understanding-openshift-pipelines.adoc
+
+[id="overview-con_{context}"]
+= Overview of {pipelines-title}
+
+{pipelines-title} is a comprehensive solution for cloud-native automation of repetitive processes, such as CI/CD.
+
+With {pipelines-shortname}, you can use custom OpenShift (Kubernetes) resources to configure your process as a set of “building blocks” that run in containers and pods on {product-title}.
+
+There are several ways to access reusable “building blocks” as well as create your own.
+
+== Steps, tasks, and pipelines
+
+The most basic “building block” of the Pipelines CI/CD process is the _step_. A step is single operation, such as running some unit tests for a Python web app, or the compilation of a Java program. Tekton performs each step in a separate container
+
+A _task_ is a collection of steps, which always run in a fixed sequential order. Pipelines runs all steps of a task as containers in the same OpenShift pod, which can provide a shared environment for the steps. For example, you can mount a volume in a task, and it is accessible inside each step of the task.
+
+A _pipeline_ is a collection of tasks. You can arrange tasks in a pipeline in a flexible order, when some of them can run in parallel as well as sequentially. You can also control execution of tasks in a pipeline in other ways, for example, specify a condition for a task to proceed, require an automatic retry if a task fails, and so on.
+
+Variable parameters are supported at every level. A step can have parameters , and a task that includes that step must set these parameters. In turn, a task can also have parameters and a pipeline that includes the task must set these parameters. A pipeline, too, can have variable parameters.
+
+.Example diagram of steps and tasks in a pipeline
+image::338_OpenShift_Pipeline_0523.png[A diagram representing a pipeline. Task A runs first and includes three steps. When task A completes, task B and task C both start; each of these tasks has two steps. Task D starts when both task B and task C complete; this task has one step.]
+
+{pipelines-shortname} specifies tasks and pipelines as custom OpenShift resources.
+
+Typically, you configure a complete CI/CD process as a pipeline, while logical parts of this process become tasks.
+
+== Task runs and pipeline runs
+
+`PipelineRun` and `TaskRun` custom resources represent individual runs of a pipeline or a task.
+
+{pipelines-shortname} automatically runs a pipeline when you create a `PipelineRun` custom resource referring to an existing pipeline. It automatically runs a task when you create a `TaskRun` custom resource referring to an existing task.
+
+The `PipelineRun` resource must set any variable parameters that the pipeline has. The `TaskRun` resource must set any variable parameters that the task has.
+
+== Triggers, trigger bindings, trigger templates, event listeners
```

Comment:
Should we can add the interceptor and it's brief details also?

Reply:
Agreed